source/server/options_impl_platform_linux.cc

```diff
@@ -0,0 +1,42 @@
+#include "server/options_impl_platform_linux.h"
+
+#include <sched.h>
+
+#include <thread>
+
+#include "common/api/os_sys_calls_impl_linux.h"
+
+#include "server/options_impl_platform.h"
+
+namespace Envoy {
+
+uint32_t OptionsImplPlatformLinux::getCpuAffinityCount(unsigned int hw_threads) {
+  unsigned int threads = 0;
+  pid_t pid = getpid();
+  cpu_set_t mask;
+  auto& linux_os_syscalls = Api::LinuxOsSysCallsSingleton::get();
+
+  CPU_ZERO(&mask);
+  const Api::SysCallIntResult result =
+      linux_os_syscalls.sched_getaffinity(pid, sizeof(cpu_set_t), &mask);
+  if (result.rc_ == -1) {
+    // Fall back to number of hardware threads.
+    return hw_threads;
```

Comment:
Can you add a test for this case? The value in adding linux syscalls is now you can have a mock and unit test this function.